credits.php

```diff
@@ -16,24 +16,41 @@
 
 echo "<h2>" . _("Bundled Software"). "</h2>";
 
-echo "<p>" . _("The following open source code is bundled and used with the dproofreaders code.") . "</p>";
+echo "<p>" . _("The following open source code is bundled with and used by the dproofreaders code.") . "</p>";
 
-$credit_details = load_credit_details($code_dir);
-asort($credit_details);
-echo "<ul>";
-foreach($credit_details as $detail)
-{
-    echo "<li>";
-    echo "<b>" . $detail["name"] . "</b><br>";
-    echo "<a href='" . $detail["url"] . "'>" . $detail["url"] . "</a><br>";
-    echo "<a href='" . $detail["license_url"] . "'>" . $detail["license"] . "</a>";
-    echo "</li>";
-}
-echo "</ul>";
+$credit_details = load_bundled_credit_details($code_dir);
+output_credit_details($credit_details);
+
+echo "<h2>" . _("Dependencies"). "</h2>";
+
+echo "<p>" . _("The following open source dependencies are used by the dproofreaders code.") . "</p>";
+
+$credit_details = load_composer_credit_details($code_dir);
+output_credit_details($credit_details);
 
 #----------------------------------------------------------------------------
 
-function load_credit_details($code_dir)
+function output_credit_details($credit_details)
+{
+    echo "<ul>";
+    foreach($credit_details as $detail)
+    {
+        echo "<li>";
+        echo "<b><a href='" . $detail["url"] . "'>" . $detail["name"] . "</a></b><br>";
+        if(isset($detail["license_url"]))
+        {
+            echo _("License") . ": <a href='" . $detail["license_url"] . "'>" . $detail["license"] . "</a>";
+        }
+        else
+        {
+            echo _("License") . ": " . $detail["license"];
```

Comment:
should we encode throughout or assume that this is trusted properly encoded data?

Reply:
That's a good catch. I will add the appropriate escaping.

Reply:
And fixed.